integration-tests/src/test_loop/utils/resharding.rs

```diff
@@ -0,0 +1,487 @@
+                            signer_id.clone(),
+                            receiver_id.clone(),
+                            &signer,
+                            0,
+                            "call_yield_create_return_promise".to_string(),
+                            yield_payload.clone(),
+                            300 * TGAS,
+                            tip.last_block_hash,
+                        );
+                        store_and_submit_tx(
+                            &node_datas,
+                            &client_account_id,
+                            &txs,
+                            &signer_id,
+                            &receiver_id,
+                            tip.height,
+                            tx,
+                        );
+                    }
+                    promise_txs_sent.set(true);
+                }
+            }
+        },
+    );
+    LoopAction::new(action_fn, succeeded)
+}
+
+// After resharding and gc-period, assert the deleted `account_id`
+// is still accessible through archival node view client (if available),
+// and it is not accessible through a regular, RPC node.
+fn check_deleted_account_availability(
+    node_datas: &[TestData],
+    test_loop_data: &mut TestLoopData,
+    archival_id: &Option<AccountId>,
+    rpc_id: &AccountId,
+    account_id: &AccountId,
+    height: u64,
+) {
+    let rpc_node_data = get_node_data(node_datas, &rpc_id);
+    let rpc_view_client_handle = rpc_node_data.view_client_sender.actor_handle();
+
+    let block_reference = BlockReference::BlockId(BlockId::Height(height));
+    let request = QueryRequest::ViewAccount { account_id: account_id.clone() };
+    let msg = Query::new(block_reference, request);
+
+    let rpc_node_result = {
+        let view_client = test_loop_data.get_mut(&rpc_view_client_handle);
+        near_async::messaging::Handler::handle(view_client, msg.clone())
+    };
+    assert_matches!(rpc_node_result, Err(GarbageCollectedBlock { .. }));
+
+    if let Some(archival_id) = archival_id {
+        let archival_node_data = get_node_data(node_datas, &archival_id);
+        let archival_view_client_handle = archival_node_data.view_client_sender.actor_handle();
+        let archival_node_result = {
+            let view_client = test_loop_data.get_mut(&archival_view_client_handle);
+            near_async::messaging::Handler::handle(view_client, msg)
+        };
+        assert_matches!(
+            archival_node_result,
+            Ok(QueryResponse { kind: QueryResponseKind::ViewAccount(_), .. })
+        );
+    }
+}
+
+// Loop action testing a scenario where a temporary account is deleted after resharding.
+// After `gc_num_epochs_to_keep epochs` we assert that the account
+// is not accesible through RPC node but it is still accesible through archival node.
+pub(crate) fn temporary_account_during_resharding(
+    archival_id: Option<AccountId>,
+    rpc_id: AccountId,
+    originator_id: AccountId,
```

Comment:
nit: What about calling this `beneficiary_id` with consistency with other existing methods? Does it help readability?

Reply:
`originator` is the creator of an account. In this case the new account is `account6.account6` and only `account6` is allowed to be originator. While in other tests, beneficiary can be anyone.

Reply:
added comment: 19a382c585b68eb81a40704fc93913c42b522dd2